rust/src/detect/transforms/subslice.rs

```diff
@@ -0,0 +1,337 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::detect::SIGMATCH_OPTIONAL_OPT;
+use suricata_sys::sys::{
+    DetectEngineCtx, DetectEngineThreadCtx, InspectionBuffer, SCDetectHelperTransformRegister,
+    SCDetectSignatureAddTransform, SCInspectionBufferCheckAndExpand, SCInspectionBufferTruncate,
+    SCTransformTableElmt, Signature,
+};
+
+use num::abs;
+use std::os::raw::{c_int, c_void};
+
+use std::ffi::CStr;
+
+static mut G_TRANSFORM_SUBSLICE_ID: c_int = 0;
+
+#[derive(Debug, PartialEq)]
+struct DetectTransformSubsliceData {
+    pub start: isize,
+    pub end: Option<isize>,
+}
+
+impl Default for DetectTransformSubsliceData {
+    fn default() -> Self {
+        DetectTransformSubsliceData {
+            start: 0,
+            end: None,
+        }
+    }
+}
+
+fn subslice_do_parse(i: &str) -> Option<DetectTransformSubsliceData> {
+    let i = i.trim().strip_prefix('[')?.strip_suffix(']')?;
+    let parts: Vec<_> = i.split(',').map(str::trim).collect();
+
+    match parts.as_slice() {
+        [start] => {
+            let start = start.parse::<isize>().ok()?;
+            Some(DetectTransformSubsliceData { start, end: None })
+        }
+        [start, end] => {
```

Comment:
think it's the first time I see this bracket notation, what does it do?

Reply:
I understand it is a slice of 2 elements, and so, we can name them

Reply:
If I understand the comments correctly, the suggestions are
1. `subslice: <offset-value> [,<nbytes-value]`
2. `subslice: offset <value> [, nbytes <value>]`
instead of
`subslice: [<offset-value> [, <nbytes-value]]`

I'm open to changing it ... either 1 or 2 is more consistent with current keywords but IMO using bracket-delimited values is consistent with specifying slices..